### PR TITLE
 apps/examples/elf: Add configs for library support in apps

### PR DESCRIPTION
**Pull request by yashwanth686007**
This patch adds config labels for different libraries for providing
support to apps. It also modifies the elf loadable apps defconfig.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [f4945f207af48f67750c378c9fd14e9986975758] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [f4945f207af48f67750c378c9fd14e9986975758] - Code Rule Check (C++) OK.

File: build/configs/imxrt1050-evk/loadable_elf_apps/defconfig [hunk 1]
```diff
@@ -706,6 +706,7 @@ CONFIG_DEBUG_ERROR=y
 # CONFIG_DEBUG_SCHED is not set
 # CONFIG_DEBUG_SYSCALL is not set
 # CONFIG_DEBUG_TASH is not set
+
```
Comment (drashti304) [inline build/configs/imxrt1050-evk/loadable_elf_apps/defconfig]: Why have you added this blank line?

Comment (yashwanth686007) [reply build/configs/imxrt1050-evk/loadable_elf_apps/defconfig]: Sry, I will revert it back.

**Comment by seinfra** [comment]
Target : [ad541f5f87667a06f73bb94ccee062c434d2abfc] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [ad541f5f87667a06f73bb94ccee062c434d2abfc] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [4ea76e843d62dd48827b96aab2409bd3dd46060b] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [4ea76e843d62dd48827b96aab2409bd3dd46060b] - Code Rule Check OK.

**Comment by sunghan-chang** [comment]
obsolete because of #3293

**Pull request closed without merging**